Common/SymbolCache.cs

```diff
@@ -115,6 +130,41 @@ public static void Clear()
             _cache = new Cache();
         }
 
+        private static Tuple<bool, Symbol, InvalidOperationException> TryGetSymbol(string ticker)
+        {
+            Symbol symbol;
+            InvalidOperationException error = null;
+            if (!_cache.TryGetSymbol(ticker, out symbol))
+            {
+                // fall-back full-text search as a back-shim for custom data symbols.
+                // permitting a user to use BTC to resolve to BTC.Bitcoin
+                var search = $"{ticker.ToUpperInvariant()}.";
+                var match = _cache.Symbols.Where(kvp => kvp.Key.StartsWith(search)).ToList();
+
+                if (match.Count == 0)
+                {
+                    // no matches
+                    error = new InvalidOperationException($"We were unable to locate the ticker '{ticker}'.");
+                }
+                else if (match.Count == 1)
+                {
+                    // exactly one match
+                    symbol = match.Single().Value;
+                }
+                else if (match.Count > 1)
+                {
+                    // too many matches
+                    error = new InvalidOperationException(
+                        "We located multiple potentially matching tickers. For custom data, be sure to " +
+                        "append a dot followed by the custom data type name. For example: 'BTC.Bitcoin'. " +
+                        $"Potential Matches: {string.Join(", ", match.Select(kvp => kvp.Key))}"
+                    );
+                }
+            }
+
+            return Tuple.Create(symbol != null, symbol, error);
```

Comment:
Personal opinion here 😛 : I'd be more inclined to just throw the exception and catch it if we want to ignore it internally, which will remove the need to return a `Tuple`

Reply:
_Ironically_ -- it was initially written that way and later refactored following self review. A quick [google](https://www.google.com/search?q=exception+for+control+flow&rlz=1C1CHBD_enUS817US817&oq=exception+for+control+flow&aqs=chrome..69i57.3471j0j7&sourceid=chrome&ie=UTF-8) will yield many examples as to why using exceptions for control flow is an anti-pattern and instead, exceptions should be reserved for truly exceptional cases that are unable to be reasonably handled by the executing code.